package-res/resources/web/vizapi/VizController.js

```diff
@@ -716,29 +716,31 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var steps = colors.length-1;
   var range = max-min;
 
+  var startColorIndex;
+  var endColorIndex;
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    startColorIndex = colors.length-1;
+    endColorIndex = startColorIndex;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var positionInRange = ((value-min)/range);
+    startColorIndex = Math.floor(positionInRange * steps);
```

Comment:
what about having: `positionInRange = ((value-min)/range)*steps`? This would economize one multiplication